### PR TITLE
update manifests for controller deployment

### PR DESCRIPTION
**Pull request by avi-biton**
This update handles both RHTAPWATCH-1208 and RHTAPWATCH-1209

Update the manifests for the controller deployment:
- update the manager manifests
- update rbac resources
- patch to add kube-rbac-proxy to the manager
- add service monitor to read the metrics

### CODE REVIEW

File: config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -7,5 +7,17 @@ resources:
...
+- auth_proxy_role_binding.yaml
+- auth_proxy_client_clusterrole.yaml
+
+# prometheus
+- prometheus_service_account.yaml
+- prometheus_viewer_role.yaml
+- prometheus_role_binding.yaml
+
+# - leader_election_role.yaml
+# - leader_election_role_binding.yaml
```
Comment (yftacherzog) [inline config/rbac/kustomization.yaml:23]: Can we remove those lines?

**Comment by avi-biton** [comment]
I removed most comments and left only the comments needed for the resources and patches used by the controller

File: config/default/kustomization.yaml [hunk 2]
```diff
@@ -24,19 +24,22 @@ resources:
...
-- path: manager_metrics_patch.yaml
-  target:
-    kind: Deployment
+# patches:
+# # # # [METRICS] The following patch will enable the metrics endpoint. Ensure that you also protect this endpoint.
+# # # # More info: https://book.kubebuilder.io/reference/metrics
+# # # # If you want to expose the metric endpoint of your controller-manager uncomment the following line.
+# - path: manager_metrics_patch.yaml
+#   target:
+#     kind: Deployment
```
[the remark this answers is not in the document]
Comment (avi-biton) [reply config/default/kustomization.yaml]: Done

File: config/rbac/kustomization.yaml (hunk 1 above)
Comment (avi-biton) [reply config/rbac/kustomization.yaml:23]: Done

File: config/default/kustomization.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (avi-biton) [reply config/default/kustomization.yaml]: The patch was used for adding `--metrics-bind-address=:8080` to the manager.
Now we are using `manager_auth_proxy_patch.yaml` to do it (and to add the kube-rbac-proxy`)

**Review by yftacherzog — COMMENTED** [review]
I think it should be ok, but I might be missing something.

**Comment by avi-biton** [comment]
> I think it should be ok, but I might be missing something.

Thanks a lot :)

**Review by yftacherzog — APPROVED** [review]

**Pull request merged**